src/v/kafka/server/server.cc

```diff
@@ -704,6 +709,12 @@ add_offsets_to_txn_handler::handle(request_context ctx, ss::smp_service_group) {
         request.decode(ctx.reader(), ctx.header().version);
         log_request(ctx.header(), request);
 
+        if (ctx.recovery_mode_enabled()) {
```

Comment:
we may also want to add the check in `init_producer_id_handler::handle`, client uses it to get a producer_id to begin a idempotency  / transaction session.

Reply:
It's there! https://github.com/redpanda-data/redpanda/pull/14236/files#diff-d66a9af30ebd2f5bc30d59587cb0f9b1d0d8ea3121a9d5411a08d69191319af4R1138